File: scripts/agent-bench/fixture-php/src/Tokenize.php
<?php

declare(strict_types=1);

namespace Corpus;

class Tokenize
{
    /**
     * Strip non-alphanumeric characters from both ends, then lowercase.
     */
    public static function normalize(string $token): string
    {
        $trimmed = preg_replace('/^[^a-zA-Z0-9]+|[^a-zA-Z0-9]+$/', '', $token);
        if ($trimmed === null) {
            $trimmed = $token;
        }

        return strtolower($trimmed);
    }

    /**
     * Split on whitespace, normalize each token, and drop empties.
     *
     * @return list<string>
     */
    public static function tokenize(string $text): array
    {
        $parts = preg_split('/\s+/', $text, -1, PREG_SPLIT_NO_EMPTY);
        if ($parts === false) {
            return [];
        }

        $tokens = [];
        foreach ($parts as $part) {
            $normalized = self::normalize($part);
            if ($normalized !== '') {
                $tokens[] = $normalized;
            }
        }

        return $tokens;
    }
}

File: scripts/agent-bench/fixture-java/src/main/java/com/corpus/Tokenize.java
package com.corpus;

import java.util.ArrayList;
import java.util.List;

public class Tokenize {
    // strip non-alphanumeric ends, lowercase
    public static String normalize(String token) {
        int start = 0;
        int end = token.length();
        while (start < end && !Character.isLetterOrDigit(token.charAt(start))) {
            start++;
        }
        while (end > start && !Character.isLetterOrDigit(token.charAt(end - 1))) {
            end--;
        }
        return token.substring(start, end).toLowerCase();
    }

    // split whitespace, normalize, drop empties
    public static List<String> tokenize(String text) {
        List<String> out = new ArrayList<>();
        for (String piece : text.split("\\s+")) {
            String norm = normalize(piece);
            if (!norm.isEmpty()) {
                out.add(norm);
            }
        }
        return out;
    }
}
